src/button-group/README.md

```diff
@@ -0,0 +1,145 @@
+# Button Group Component
+
+Groups a collection of buttons together. Can also operate as a radio group or checkbox group.
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button, StatefulButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <StatefulButtonGroup>
+      <Button>Copy</Button>
+      <Button>Cut</Button>
+      <Button>Paste</Button>
+    </StatefulButtonGroup>
+}
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {Button, ButtonGroup} from 'baseui/button-group';
+
+export default class MyCustomLogic extends React.Component {
+  state = {selected: [0, 1]};
+
+  handleClick = (event, index) => {
+    if (!this.state.selected.includes(index)) {
+      this.setState({selected: [...this.state.selected, index]});
+    } else {
+      this.setState({
+        selected: this.state.selected.filter(value => value !== index),
+      });
+    }
+  }
+
+  handleBold = (event) => {
+    // event handler only for the 'bold' button
+  }
+
+  render() {
+    return (
+      <ButtonGroup
+        selected={this.state.selected}
+        mode="checkbox"
+        onClick={this.handleClick}
+      >
+        <Button onClick={this.handleBold}>Bold</Button>
+        <Button>Italic</Button>
+        <Button>Underline</Button>
+      </ButtonGroup>
+    )
+  }
+}
+```
+
+## Exports
+
+* `Button`
+* `ButtonGroup`
+* `StatefulButtonGroup`
+* `StatefulContainer`
+* `StyledButton`
+* `StyledRoot`
+* `MODE`
+* `STATE_CHANGE_TYPE`
+
+## `Button` API
+
+Same as `{Button} from baseui/button`, with additional properties:
+* `first?: boolean = false`
+  * Removes right corner radii from button.
+* `last?: boolean = false`
+  * Removes left corner radii from button.
```

Comment:
As a customer it would be nice if this was done magically – would it possible to apply `:first-child, :last-child { border-radius: 0 }` to Button when used in a ButtonGroup?

Reply:
I think it’s being already done magically via cloneElement. Imo much better than CSS selectors since it preserves component isolation.

Reply:
we could possibly build the `selected` state into the core button, and remove this "extension" component entirely. i think that could better experience for devs. does the code below look reasonable, or more complicated?

```js
import React from 'react';
import {Button} from 'baseui/button';
import {ButtonGroup} from 'baseui/button-group'
```

Reply:
^Yea, that’s a good idea!

Reply:
🙆 cool, i'll take some time today to rework it

Reply:
implemented with overrides... could not get pseudo selectors working